owners/src/owners.js

```diff
@@ -219,6 +219,55 @@ class OwnersRule {
   }
 }
 
+/**
+ * A pattern-based ownership rule applying to all matching files in the
+ * directory and all subdirectories.
+ *
+ * Treats `*` as a wildcard glob pattern; all other characters are treated as
+ * literals.
+ */
+class PatternOwnersRule extends OwnersRule {
+  /**
+   * Constructor.
+   *
+   * @param {!string} ownersPath path to OWNERS file.
+   * @param {string[]} owners list of GitHub usernames of owners.
+   * @param {!string} pattern filename/type pattern.
+   */
+  constructor(ownersPath, owners, pattern) {
+    super(ownersPath, owners);
+    this.pattern = pattern;
+    this.regex = new RegExp(
+      pattern
+        .split('*')
+        .map(PatternOwnersRule.escapeRegexChars)
+        .join('.*?')
+        .split(/\s*,\s*/)
+        .join('|')
```

Comment:
This is kind of cryptic... I wonder if using something like [minimatch](https://www.npmjs.com/package/minimatch) or [glob](https://www.npmjs.com/package/glob) instead of this whole shebang would make sense

Reply:
The bottom two lines here disappear in a later, the comma bit belongs in the parser, which just leaves `*` as the only thing being replaced.

Reply:
Raised discussion in #278